python-lib/azure_client.py

```diff
@@ -69,6 +72,17 @@ def __init__(self, project_key, config):
         # Connect to Graph API
         self.set_session_headers()
 
+    def assert_valid_login_remapping(self):
+        for login_remapping in self.login_remapping:
+            print("ALX:login_remapping={}".format(login_remapping))
+            to_value = login_remapping.get("to")
+            if to_value and not self.is_valid_login(to_value):
+                raise Exception("'{}' in the login remapping is not a valid character for a DSS user login".format(to_value))
```

Comment:
maybe add the allowed characters to the message, for example "valid characters must match the regex pattern [a-zA-Z0-9@.+_-]"

Reply:
👍